src/RecordData/Logger.cpp

```diff
@@ -0,0 +1,149 @@
+#include "Logger.h"
+
+using namespace mmfs;
+
+static const char *logTypeStrings[] = {"LOG", "ERROR", "WARNING", "INFO"};
+static Mode mode = GROUND;
+
+Logger::Logger(GroundDest groundDest = ALTERNATING_BOTH, uint16_t bufferSize = 25000, int bufferInterval = 300) 
+{
+    ram = new PSRAM();
+    dataFile = new SdCardFile();
+    logFile = new SdCardFile();
+    ready = false;
+    dumped = false;
+
+    if (ram->init() && dataFile->init("Log.txt") && logFile->init("FlightData.csv"))
```

Comment:
wait does the log file have to be named this? Like what happens if you run the board twice? Does it overwrite the old FlightData.csv?

Reply:
Prob fine to just like while loop through FlightData_i until you get to one that doesn't exist

Reply:
I agree, but naming them ###_FlightData might be better as it keeps the log and flight file together, sorted by flight number. The way you mentioned sorts by # but separates the log file from the flight file. We could also use folders, but I think together makes things easier.

Reply:
it does it in the same format as before, where it searches the SD card and finds the latest numbered file and increments it to create a new one. The parameter passed in is just the file name "suffix" thats passed in after the `###_`